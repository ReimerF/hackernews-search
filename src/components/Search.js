import React from "react";
import PropTypes from "prop-types";

const Search = ({ onSearchChange, onSearchSubmit, searchTerm, children }) => (
  <div>
    <form onSubmit={onSearchSubmit}>
      <button type="submit">{children} ></button>
      <input type="text" onChange={onSearchChange} value={searchTerm} />
    </form>
  </div>
);

Search.propTypes = {
  onSearchChange: PropTypes.func.isRequired,
  onSearchSubmit: PropTypes.func.isRequired,
  searchTerm: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
};

export default Search;
