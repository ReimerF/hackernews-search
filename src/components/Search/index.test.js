import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Search from "./index";

describe("Search", () => {
  it("renders Search without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Search
        onSearchChange={jest.fn()}
        onSearchSubmit={jest.fn()}
        searchTerm="test"
      >
        Search
      </Search>,
      div
    );
  });

  test("Search has a valid snapshot", () => {
    const component = renderer.create(
      <Search
        onSearchChange={jest.fn()}
        onSearchSubmit={jest.fn()}
        searchTerm="test"
      >
        Search
      </Search>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
