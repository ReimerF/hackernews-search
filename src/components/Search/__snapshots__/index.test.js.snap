// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Search has a valid snapshot 1`] = `
<div>
  <form
    onSubmit={[MockFunction]}
  >
    <button
      type="submit"
    >
      Search
       &gt;
    </button>
    <input
      onChange={[MockFunction]}
      type="text"
      value="test"
    />
  </form>
</div>
`;
